ignite/contrib/metrics/regression/canberra_metric.py

```diff
@@ -22,13 +25,21 @@ class CanberraMetric(_BaseRegression):
 
     """
 
+    def __init__(
+        self, output_transform: Callable = lambda x: x, device: Union[str, torch.device] = torch.device("cpu")
+    ):
+        self._sum_of_errors = None
+        super(CanberraMetric, self).__init__(output_transform, device)
+
+    @reinit__is_reduced
     def reset(self):
-        self._sum_of_errors = 0.0
+        self._sum_of_errors = 0
 
     def _update(self, output):
         y_pred, y = output
```

Comment:
We have to detach the output too as it is done for core metrics

Reply:
@vfdev-5 done